lib/events-reader.js

```diff
@@ -105,95 +109,124 @@ module.exports = function (harvesterApp) {
 
                 debug('got data from oplog ' + JSON.stringify(doc) + ' ts: ' + logTs(doc.ts));
 
-                _.forOwn(harvesterApp.changeHandlers, function (changeHandlers, key) {
+                var matchedChangeHandlers = matchChangeChandlers(harvesterApp.changeHandlers, doc.ns);
+                _.forEach(matchedChangeHandlers, function (changeHandler) {
 
-                    _.forEach(changeHandlers, function (changeHandler) {
 
-                        var resource = inflect.pluralize(key);
-                        var regex = new RegExp('.*\\.' + resource, 'i');
-                        var match = regex.test(doc.ns);
+                        var asyncInMemory = changeHandler['asyncInMemory'];
 
-                        var dfd = RSVP.defer();
+                        if (!asyncInMemory) {
+                            var dfd = RSVP.defer();
+                            promises.push(dfd.promise);
+                            processWithHandlerT(that, changeHandler, doc, dfd);
```

Comment:
yes, it's been in there for a long time, not just for this change 
usually it's not a good practice to use deferreds 
but in this case it fits